spyder/workers/tests/test_update.py

```diff
@@ -6,56 +6,133 @@
+    )
+
+    if is_anaconda:
+        if is_pypi:
+            channel = ("pypi", "https://conda.anaconda.org/pypi")
+        else:
+            channel = spyder_conda_channel
+
+        mocker.patch(
+            "spyder.workers.updates.get_spyder_conda_channel",
+            return_value=channel
+        )
+
+    worker = WorkerUpdates(None, False, version="1.0.0")
+    worker.start()
+    assert worker.update_available
+    assert len(worker.releases) == 1
+
+
+@pytest.mark.parametrize("is_anaconda", [True, False])
+@pytest.mark.parametrize("is_pypi", [True, False])
+@pytest.mark.parametrize(
+    "spyder_conda_channel", [
+        ("pkgs/main", "https://repo.anaconda.com/pkgs/main"),
+        ("conda-forge", "https://conda.anaconda.org/conda-forge")
+    ]
+)
+def test_no_update(qtbot, mocker, is_anaconda, is_pypi, spyder_conda_channel):
+    """Test we don't offer updates for Anaconda and PyPI."""
+    mocker.patch(
+        "spyder.workers.updates.is_anaconda",
+        return_value=is_anaconda
+    )
+
+    if is_anaconda:
+        if is_pypi:
+            channel = ("pypi", "https://conda.anaconda.org/pypi")
+        else:
+            channel = spyder_conda_channel
+
+        mocker.patch(
+            "spyder.workers.updates.get_spyder_conda_channel",
+            return_value=channel
+        )
+
+    worker = WorkerUpdates(None, False, version="1000.0.0")
+    worker.start()
+    assert not worker.update_available
+    assert len(worker.releases) == 1
+
+
+def test_update_installers(qtbot, mocker):
+    """Test we offer updates for our installers."""
+    mocker.patch("spyder.workers.updates.is_anaconda", return_value=False)
+    mocker.patch("spyder.workers.updates.is_pynsist", return_value=True)
+
     worker = WorkerUpdates(None, False, version="1.0.0")
     worker.start()
     assert worker.update_available
+    assert len(worker.releases) > 1
+
 
+def test_no_update_installers(qtbot, mocker):
+    """Test we don't offer updates for our installers."""
+    mocker.patch("spyder.workers.updates.is_anaconda", return_value=False)
+    mocker.patch("spyder.workers.updates.is_pynsist", return_value=True)
 
-def test_no_update(qtbot):
-    """Test we don't offer updates for very high versions."""
     worker = WorkerUpdates(None, False, version="1000.0.0")
     worker.start()
     assert not worker.update_available
+    assert len(worker.releases) > 1
 
 
-def test_no_update_development(qtbot):
+def test_no_update_development(qtbot, mocker):
     """Test we don't offer updates for development versions."""
+    mocker.patch(
+        "spyder.workers.updates.get_spyder_conda_channel",
+        return_value=("pypi", "https://conda.anaconda.org/pypi")
+    )
+
     worker = WorkerUpdates(None, False, version="3.3.2.dev0",
                            releases=['3.3.1'])
     worker.start()
     assert not worker.update_available
 
 
-def test_update_pre_to_pre(qtbot):
+def test_update_pre_to_pre(qtbot, mocker):
```

Comment:
Should we make `version` a pytest parameter here also?

Reply:
I prefer to leave this and the test below without parametrizing because they are easier to understand.